src/shaders/circle.vertex.glsl

```diff
@@ -102,10 +137,13 @@ void main(void) {
             view_scale = projected_center.w;
         #endif
     #endif
+    #if defined(SCALE_WITH_MAP) && defined(PROJECTION_GLOBE_VIEW)
+        view_scale *= a_scale;
+    #endif
     gl_Position = project_vertex(extrude, world_center, projected_center, radius, stroke_width, view_scale);
```

Comment:
`project_vertex` could take `globe_surface_extrusion` as parameter to avoid some redundant calculation of `extrudeOnGlobeSurface`?

Reply:
I've been thinking about that but inside `project_vertex` we use a different extrusion value. Maybe I could return a matrix from `extrude_on_globe_surface` and pass that around to do the transformation of the extrusion.